gwpy/timeseries/timeseries.py

```diff
@@ -1931,14 +1931,15 @@ def q_transform(self, qrange=(4, 64), frange=(0, numpy.inf),
 
         # build regular Spectrogram from peak-Q data by interpolating each
         # (Q, frequency) `TimeSeries` to have the same time resolution
-        nx = int(abs(Segment(*outseg)) / tres)
+        xout = numpy.arange(outseg[0], outseg[1], tres)
+        nx = xout.size
+
```

Comment:
What's the motivation for this change? Does this not give the same answer as before?

Reply:
Sorry, I thought I replied to this already.
It turns out that when tres is calculated and is not an integer multiple/divisor of the sample frequency that we can end up with round  off errors that result in a difference of 1.

Reply:
@areeda, I've also addressed this as part of [**#984**](https://github.com/gwpy/gwpy/blob/7a3ab3183d138e6c1442f11380ffe56d1363b630/gwpy/signal/qtransform.py#L526-L530) which will end up superseding this since that PR refactors spectrogram interpolation into `gwpy.signal.qtransform`.